services/storage/src/simcore_service_storage/dsm_factory.py

```diff
@@ -50,7 +51,12 @@ async def list_files_in_dataset(
 
     @abstractmethod
     async def list_files(
-        self, user_id: UserID, *, expand_dirs: bool, uuid_filter: str = ""
+        self,
+        user_id: UserID,
+        *,
+        expand_dirs: bool,
+        uuid_filter: str = "",
+        project_id: ProjectID | None = None,
     ) -> list[FileMetaData]:
         """returns all the file meta data a user has access to (uuid_filter may be used)"""
```

Comment:
I think these comments do not make sense anymore with your change

Reply:
I have corrected these comments so they make sense also with my changes.